parsec/core/gui/instance_widget.py

```diff
@@ -212,13 +212,20 @@ def login_with_password(self, key_file, password):
 
     def show_central_widget(self):
         self.clear_widgets()
-        central_widget = CentralWidget(
-            self.core,
-            self.core_jobs_ctx,
-            self.core.event_bus,
-            systray_notification=self.systray_notification,
-            parent=self,
-        )
+        try:
+            central_widget = CentralWidget(
+                self.core,
+                self.core_jobs_ctx,
+                self.core.event_bus,
+                systray_notification=self.systray_notification,
+                parent=self,
+            )
+        except AttributeError:
+            # The core can be set to None at any time if do_run_core get an error, is cancelled or
```

Comment:
I think we should take some time to think about the states of this `InstanceWidget`

The fact that `self.core` can be set to `None` at any time makes things much more complicated.
For instance this code is not thread safe given the check on self.core not None is done in a different step than the access: https://github.com/Scille/parsec-cloud/blob/710ca19c946bface10616637604c862b218c3210/parsec/core/gui/instance_widget.py#L82-L85

On top of that we could have subtle error in:
https://github.com/Scille/parsec-cloud/blob/710ca19c946bface10616637604c862b218c3210/parsec/core/gui/instance_widget.py#L217-L218

If `self.core` gets executed after `self.core.event_bus` (for instance if we refactor the code following my previous comment ^^), we can end up with `self.core != None` when we access `self.core.event_bus`, then `self.core == None` when we retrieve the core...

So I wonder if we shouldn't avoid entirely to reset `self.core`. Considering the tab containing the `InstanceWidget` is going to be closed, we should be able to state that this class cannot be re-used (so it lifespan would be `creation -> login -> error`, `creation -> login -> ready -> error` or `creation -> login -> ready -> logout`).

Reply:
```
Considering the tab containing the InstanceWidget is going to be closed,
```
This is not the case when user logs out. We could change the logout mecanism.